### ISSUE TITLE
[Bug Report] Mail Server fails to send e-mails when using SQL Server database backend.

### ISSUE THREAD

**Issue by abjoseph** [documize/community#298]
## Title:

Mail Server smtp configuration fails to send any e-mails when the underlying database being used is SQL Server.

## Functional Impact:

Not able to leverage/utilize e-mail based functionality within Documize instance such as email notifications or password reset.

## Minimal Repro Steps:

 1. Provision an instace of SQL Server 2016+ (Standard)
 2. Download either the [Community Edition](https://github.com/documize/community/releases) or [Enterprise Edition](https://www.documize.com/downloads)
 3. Create a toml base configuration based on the provided SQL Server based [template](https://docs.documize.com/api/public/attachment/8SwAtYoB/bj3ggv67m02fu20qv1qg).
 4. Start documize executable using the toml configuration from #3.
	1. **Linux** - ``` ./documize config.conf ```
	2. **Windows** - ``` .\documize.exe config.conf ```
 5. Navigate to ``` Hostname/Ip:5001 ``` and go to *Mail Server* configuration section within the *Settings* top-level menu.
 6. Enter the relevant smtp settings, issues persist regardless of the specific mail settings. Gmail smtp was used my testing:

```
• Gmail SMTP server address: smtp.gmail.com
• Gmail SMTP username: Your full Gmail address (e.g. yourusername@gmail.com)
• Gmail SMTP password: Your Gmail password
• Gmail SMTP port (TLS): 587
• Gmail SMTP port (SSL): 465
• Gmail SMTP TLS/SSL required: false(TLS)/true(SSL)
```
\*\*Be sure to turn on "*[less secure app access](https://devanswers.co/allow-less-secure-apps-access-gmail-account/)*" in gmail settings, for testing purposes only!

 7. Click Save & Test

## Expected Result:

- **Should see two alert notifications:**
	1. *Sending test email to you*
    2. *Email sent successfully!*

![mail test - success](https://user-images.githubusercontent.com/26844547/61186515-67b5f900-a634-11e9-9621-4a63aedc0805.png)

## Actual Result:

- **Got two alert notifications:**
	1. *Sending test email to you*
    2. *Unable to send test email: dial tcp :0: connect: connection refused*

![mail test - fail](https://user-images.githubusercontent.com/26844547/61186514-5c62cd80-a634-11e9-97c3-86b18490b0ce.png)

### Informational -

The above setup was reproduced utilizing **postgres** database instead of Sql Server with identical mail settings and the Expected Results were observed.

### Technical -

- Documize Version: Enterprise Edition v3.1.0 (build 190629131646)
- Platforms: 
	1. *Ubuntu 18.04.2 LTS*
	2. *Windows 10 Enterprise - version 1809*
- User:
	1. *Linux -* Non-Root user with Sudo permissions.
	2. *Windows -* Regular User with Administrator permissions.
- SQL Server:
	- *Version:* ``` Microsoft SQL Server 2017 (RTM) - 14.0.1000.169 (X64)   Aug 22 2017 17:04:49   Copyright (C) 2017 Microsoft Corporation  Standard Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor) ```
	- *Collation:* ``` SQL_Latin1_General_CP437_CI_AS ```...[Reference](https://database.guide/what-is-collation-in-databases/)

### \*\*EDIT: 
*It appears that any information being read from the dbo.dmz_config table when using SQL Server as a backend is affected. Reason being, Trello integration also fails for SQL Server, the App Key value for which comes from the dbo.dmz_config table as well.*

**Comment by danatkdan** [comment]
I can confirm the same issue with our instance of Documize 3.1.0 on MS SQL 2016.

**Comment by HarveyKandola** [comment]
We're looking into this at the moment...

**Comment by HarveyKandola** [comment]
Fixed for 3.1.1 which lands momentarily.

**Comment by abjoseph** [comment]
@HarveyKandola Thanks
